### PR TITLE
(maint) Update tags for PA 1.4.0

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

File: configs/components/windows_puppet.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
   "url": "git://github.delivery.puppetlabs.net/puppetlabs-puppet_for_the_win.git",
-  "ref": "refs/tags/4.4.0.1"
+  "ref": "refs/tags/4.4.0"
```
Comment (melissa) [inline configs/components/windows_puppet.json]: @Iristyle changes per your comment in https://tickets.puppetlabs.com/browse/PA-205

File: configs/components/windows_ruby.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "url": "git://github.delivery.puppetlabs.net/puppetlabs-puppet-win32-ruby.git",
   "ref": {
-    "x86": "refs/tags/2.1.8.2-x86",
-    "x64": "refs/tags/2.1.8.2-x64"
+    "x86": "refs/tags/2.1.8.1-x86",
+    "x64": "refs/tags/2.1.8.1-x64"
```
Comment (melissa) [inline configs/components/windows_ruby.json]: @Iristyle changed per your comment in https://tickets.puppetlabs.com/browse/PA-205

File: configs/components/hiera.json [hunk 3]
```diff
@@ -1 +1 @@
-{"url": "git://github.com/puppetlabs/hiera.git", "ref": "3047016da65d12ff7e316db17e40adec9a080090"}
+{"url": "git://github.com/puppetlabs/hiera.git", "ref": "refs/tags/3.1.0"}
```
Comment (melissa) [inline configs/components/hiera.json]: The tag is off `fc7614d6ba81845757ba7318269fad2b2da08da3`. `3047016da65d12ff7e316db17e40adec9a080090` doesn't seem to exist in the hiera repo... So I'd like some validation on this tag. It also looks like the latest hiera tag is `3.1.5`.

cc/ @kylog

Comment (kylog) [inline configs/components/hiera.json]: Re tags: 3.1.0 is correct. There is no 3.1.5 tag in hiera (although there is in facter).

Re commits: https://github.com/puppetlabs/hiera/commit/3047016da65d12ff7e316db17e40adec9a080090 is in the hiera repo, and is included in hiera's 3.1.0 tag.

Comment (melissa) [inline configs/components/hiera.json]: oops! I had accidentally fetched the facter repo in my hiera repo! Thanks, Kylo! With that fixed, the 3.1.0 tag does correspond with the `3047016da65d12ff7e316db17e40adec9a080090` sha :) yay!

File: configs/components/windows_puppet.json (hunk 1 above)
Comment (joshcooper) [inline configs/components/windows_puppet.json]: I think @Iristyle's comment is outdated. That should be 4.4.0.1 to pick up the change for https://tickets.puppetlabs.com/browse/PA-225 which is to be released in puppet-agent 1.4.0.

File: configs/components/windows_ruby.json (hunk 2 above)
Comment (joshcooper) [inline configs/components/windows_ruby.json]: I think @Iristyle's comment was made before we decided to bump to openssl 1.0.2g in https://tickets.puppetlabs.com/browse/RE-6662, first released in puppet-agent 1.3.6 (and should be included in 1.4.0).

Comment (melissa) [inline configs/components/windows_ruby.json]: Updated. Thanks @joshcooper

File: configs/components/windows_puppet.json (hunk 1 above)
Comment (melissa) [inline configs/components/windows_puppet.json]: Updated. Thanks @joshcooper

**Comment by McdonaldSeanp** [comment]
:+1:

**Pull request merged**